api/api-pipeline-selection-v1/src/main/java/com/thoughtworks/go/apiv1/pipelineselection/representers/PipelineSelectionResponse.java

```diff
@@ -17,21 +17,26 @@
 package com.thoughtworks.go.apiv1.pipelineselection.representers;
 
 import com.thoughtworks.go.config.PipelineConfigs;
-import com.thoughtworks.go.server.domain.user.PipelineSelections;
 
 import java.util.List;
 
 public class PipelineSelectionResponse {
-    private final PipelineSelections selectedPipelines;
+    private final List<String> pipelines;
+    private final boolean blacklist;
     private final List<PipelineConfigs> pipelineConfigs;
 
-    public PipelineSelectionResponse(PipelineSelections selectedPipelines, List<PipelineConfigs> pipelineConfigs) {
-        this.selectedPipelines = selectedPipelines;
+    public PipelineSelectionResponse(List<String> pipelines, boolean blacklist, List<PipelineConfigs> pipelineConfigs) {
+        this.pipelines = pipelines;
+        this.blacklist = blacklist;
         this.pipelineConfigs = pipelineConfigs;
     }
 
-    public PipelineSelections getSelectedPipelines() {
-        return selectedPipelines;
+    public List<String> selections() {
+        return pipelines;
+    }
+
+    public boolean blacklist() {
+        return blacklist;
```

Comment:
Avoid needing to construct a `PipelineSelections` instance just to be able to de/serialize. Makes testing easier, and the usage clearer during controller actions.